client/js/renderPreview.js

```diff
@@ -61,3 +61,56 @@ function handleImageInPreview(content, container) {
 		});
 	}
 }
+
+/* Image viewer */
+
+// FIXME Remove #input focus when this is open
+$("#viewport").on("click", ".toggle-content img", function() {
+	const self = $(this);
+	const link = self.closest("a");
+
+	openImageViewer(self, link);
+
+	// Prevent the link to open a new page since we're opening the image viewer,
+	// but keep it a link to allow for Ctrl/Cmd+click
+	event.preventDefault();
+});
+
+// Reason to make this a specific handler is, for images, to open the image
+// viewer even when clicking in the margins of the thumbnail
+$("#viewport").on("click", ".toggle-content.toggle-type-image", function() {
+	const self = $(this);
+	const image = self.find("img");
+
+	openImageViewer(image, self);
+
+	return false;
+});
+
+$(document).on("click", "#image-viewer", function() {
+	closeImageViewer();
+});
+
+$(document).keydown(function(e) {
+	switch (e.keyCode ? e.keyCode : e.which) {
+	case 27: // Escape
```

Comment:
I don't really want to get into keyboard handling as part of this PR. Can we do it later?
(Also, I'm not 100% sure we should, hesitating a bit, let's postpone this I'd say)